xbmc/addons/kodi-dev-kit/include/kodi/AddonBase.h

```diff
@@ -139,39 +139,75 @@ class ATTRIBUTE_HIDDEN CSettingValue
   ///@{
 
   /// @brief To get settings value as string.
-  std::string GetString() const { return (const char*)m_settingValue; }
+  std::string GetString() const { return str; }
 
   /// @brief To get settings value as integer.
-  int GetInt() const { return *(const int*)m_settingValue; }
+  int GetInt() const { return atoi(str.c_str()); }
 
   /// @brief To get settings value as unsigned integer.
-  unsigned int GetUInt() const { return *(const unsigned int*)m_settingValue; }
+  unsigned int GetUInt() const { return atoi(str.c_str()); }
 
   /// @brief To get settings value as boolean.
-  bool GetBoolean() const { return *(const bool*)m_settingValue; }
+  bool GetBoolean() const { return atoi(str.c_str()) > 0; }
 
   /// @brief To get settings value as floating point.
-  float GetFloat() const { return *(const float*)m_settingValue; }
+  float GetFloat() const { return static_cast<float>(atof(str.c_str())); }
 
   /// @brief To get settings value as enum.
   /// @note Inside settings.xml them stored as integer.
   template<typename enumType>
   enumType GetEnum() const
   {
-    return static_cast<enumType>(*(const int*)m_settingValue);
+    return static_cast<enumType>(GetInt());
   }
 
   ///@}
 
 private:
-  const void* m_settingValue;
+  const std::string str;
 };
 ///@}
 //------------------------------------------------------------------------------
 
 namespace addon
 {
 
+struct ATTR_DLL_LOCAL CPrivateBase
+{
+  /* Global variables of class */
+  static AddonGlobalInterface*
```

Comment:
would it make sense to switch to`std::unique_ptr`here. I have seen so many manual deletes in so many files....